src/models/solution.rs

```diff
@@ -0,0 +1,31 @@
+use serde::{Deserialize, Serialize};
+use surrealdb::sql::Thing;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Solution {
+    pub id: Option<Thing>,
+
+    pub problem_id: Thing,
```

Comment:
```suggestion
    pub problem: Thing,
```

Consider use `problem` instead?